### PR TITLE
feat(dashboard): created UI for viewing courses on the dashboard

### PR DESCRIPTION
**Pull request by mitchfaulcon**
### Purpose

Implemented the frontend for the dashboard page for viewing courses

[HRT-64](https://softeng761team5.atlassian.net/browse/HRT-64)

### Approach

- Uses the dummy course data to populate a flexbox of course cards
- These can be clicked on to be redirected after the backend is hooked up
- Added a create course button that is always at the end of the course cards in the flexbox
- **Note: There is a bug where the cards don't stay centered when shrinking the window horizontally. I'm not sure how to fix it but if anyone wants to have a look and let me know that'd be great :)**

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI before this PR

image

#### Screenshot(s) of UI after this PR

image
image

### CODE REVIEW

**Review by preetpatel — DISMISSED** [review]

File: src/components/Course.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { Card, Typography } from "antd";
+import "../styles/Course.css";
+
+interface CourseProps {
+  courseName: string;
+}
+
+const Dimension: React.FC<CourseProps> = (props: CourseProps) => {
+  const onClick = () => {
+    // TODO: Redirect to appropriate course page
```
Comment (gargi-bansal) [inline src/components/Course.tsx:12]: Is this going to be a different ticket (the course page on the dashboard)? It's not currently on the Jira backlog

Comment (mitchfaulcon) [reply src/components/Course.tsx:12]: Yeah I think it is, I'll add a new ticket for it

File: src/pages/Dashboard.tsx [hunk 2]
```diff
@@ -23,15 +27,38 @@ const Dashboard: React.FC<RouteComponentProps> = (props) => {
...
       <div className="Dashboard-Content">
-        <Button onClick={onInstructionsClick}>How to play</Button>
-        <Button onClick={createNewCourse}>Create new course</Button>
+        <Typography className="Courses-Heading">Courses</Typography>
+        <Button
+          type="primary"
+          className="Instructions-Button"
+          onClick={onInstructionsClick}
+        >
+          How to play
```
Comment (gargi-bansal) [inline src/pages/Dashboard.tsx:57]: I think this button now needs a shadow since it is on the main background

**Review by gargi-bansal — CHANGES_REQUESTED** [review]
Dashboard looks great! Just one small request. Can the courses be centered, it looks weird when I minimise the window a bit.
Screen Shot 2020-10-01 at 11 43 14 AM

**Review by gargi-bansal — DISMISSED** [review]

File: src/pages/Dashboard.tsx [hunk 3]
```diff
@@ -1,37 +1,84 @@
...
+        <Row
+          className="Course-Row"
+          style={{
+            marginLeft: dashboardCardsMargin,
+            marginRight: dashboardCardsMargin,
+          }}
+        >
+          {courses.map((item) => {
+            return (
+              <Course
```
Comment (preetpatel) [inline src/pages/Dashboard.tsx:74]: right here.. you need to give that course an ID.. so you can just pull it from `item._id`

Comment (mitchfaulcon) [reply src/pages/Dashboard.tsx:74]: using the key property?

Comment (preetpatel) [reply src/pages/Dashboard.tsx:74]: yup

Comment (gargi-bansal) [reply src/pages/Dashboard.tsx:74]: yup

Comment (mitchfaulcon) [reply src/pages/Dashboard.tsx:74]: thanks :)

**Review by gargi-bansal — DISMISSED** [review]

**Review by gargi-bansal — APPROVED** [review]

**Review by preetpatel — APPROVED** [review]

**Pull request merged**